### DISCUSSION TITLE
'Rare' use case and Csla

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#1777] [Questions]
Hi All,
This has happened to me once in a while in the past and it happened today : ). I have a Csla based class in one of my projects, nothing complicated, in 'Data Access' operation(s) the object is calling stored procs for 'Select', Insert etc. So far so good, now I have a scenario where the properties and business rules will stay the same but I have to call another set of stored procs, the new and 'different' stored procs have the same 'interface' (accepts and returns the same parameters).
If I try to use the same Csla based object, I have to somehow overload the 'logical' side DataPortal methods which can only be done by varying the types/number of parameters (since the names of those methods are 'fixed' in Csla).
I could use some kind of 'dummy' criteria object to fix this but is there a cleaner way?
Kind Regards

**Comment by rockfordlhotka** [comment]
What is the determining factor for selecting sproc set A or B?

**Reply by Chicagoan2016** [reply]
@rockfordlhotka, a read-only collection that shows either a) immediate 'subordinate' employees or b) all 'subordinate' employees. The two factory methods need an EmployeeID but  are called from two buttons on UI. I had thought of adding an extra property to differentiate between the two but it seemed similar to creating empty criteria objects, just got me thinking if there were a cleaner solution.
Kind Regards

**Edit:** The read-only collection is a separate use case from what I asked in the original question but both are those rare use cases that show up once in while

**Reply by rockfordlhotka** [reply]
This sort of thing is why we now allow multiple parameters to the data portal, so you can easily pass more than one 'criteria' value to your data portal operation methods.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  I was thinking about this and wondered if we could use inheritance, have a base class with all the properties and business rules (if applicable) and then each 'derived' object will have its own implementation of 'data access' methods?
Kind Regards

**Reply by rockfordlhotka** [reply]
I'd think that would work fine.
